pilot/pkg/model/push_context.go

```diff
@@ -1608,30 +1608,30 @@ func (ps *PushContext) mergeGateways(proxy *Proxy) *MergedGateway {
 
 // pre computes gateways for each network
 func (ps *PushContext) initMeshNetworks() {
-	if ps.Networks == nil || len(ps.Networks.Networks) == 0 {
-		return
-	}
-
 	ps.networkGateways = map[string][]*Gateway{}
-	for network, networkConf := range ps.Networks.Networks {
-		gws := networkConf.Gateways
-		if len(gws) == 0 {
-			// all endpoints in this network are reachable directly from others. nothing to do.
-			continue
-		}
 
-		registryNames := getNetworkRegistries(networkConf)
-		gateways := make([]*Gateway, 0, len(gws))
+	// First, if a gateway address is provided in the configuration use it. If the gateway address
+	// in the config was a hostname it got already resolved and replaced with an IP address
```

Comment:
That's not ideal - in many cases the IP may change ( global LB, etc ). Can you add a TODO and/or issue ? 

We could periodically lookup at least - but the correct solution is to do the lookups from the client side ( since they may resolve very differently ). We will have this capability with the xds proxy ( local transformations ), but ideally this would be done in Envoy, i.e. allow passing hostnames. 

@howardjohn  could we generate a cluster type DNS in this case - but still assign weights ? We need to think more, I think it'll be a common case.

Reply:
Not sure this comment is accurate... it's some copypaste from moved code. I think what this was saying was that we resolve the LoadBalancer address using the hostname in meshNetworks to find that Service.